pkg/k8shandler/status.go

```diff
@@ -47,15 +45,55 @@ func updateNodeStatus(node *nodeState, dpl *v1alpha1.ElasticsearchStatus) {
 	if node.Actual.StatefulSet != nil {
 		nodeStatus.StatefulSetName = node.Actual.StatefulSet.Name
 	}
+
+	if node.Desired.Roles != nil {
+		nodeStatus.Roles = node.Desired.Roles
+	}
 	dpl.Nodes = append(dpl.Nodes, nodeStatus)
 }
 
-func AddStatusMessage(message string, dpl *v1alpha1.Elasticsearch) error {
-	dpl.Status.Message = message
-	err := sdk.Update(dpl)
+func clusterHealth(dpl *v1alpha1.Elasticsearch) string {
+	pods, err := listPods(dpl.Name, dpl.Namespace)
 	if err != nil {
-		return fmt.Errorf("failed to update Elasticsearch status: %v", err)
+		return healthUnknown
 	}
 
-	return nil
+	// no elasticsearch pods were found
+	if len(pods.Items) == 0 {
+		return healthUnknown
+	}
+
+	// use arbitrary pod
+	pod := pods.Items[0]
+
+	config := &ExecConfig{
+		pod:            &pod,
+		containerName:  "elasticsearch",
+		command:        []string{"es_util", "--query=_cluster/health?pretty=true"},
```

Comment:
Is there a reason this is 'pretty'?  Does formatting get preserved in the status?

Reply:
The response is structured as opposite to plain text, which makes it easier to parse the response.

Reply:
doesn't this get returned as a json blob if we do `pretty` or not?
It seems golang has a package to deal with json: https://blog.golang.org/json-and-go

Reply:
In fact we already have an example in the logging code of using go to parse json: https://github.com/openshift/origin-aggregated-logging/blob/master/hack/testing/check-logs.go#L84

Reply:
Yes, reading structured data is easy in go
https://github.com/openshift/elasticsearch-operator/pull/19/files/05d4c56070c59318ed7e11ab72bbc9d93204d75a#diff-4fd53da9eef5fbd6aa1bb77f37b72b22R88